server/src/services/desks.service.ts

```diff
@@ -53,6 +59,59 @@ export function getDesksService({
   ];
 }
 
+export async function getDesksServiceReal({
+  zoneId,
+  deskIds = [],
+  count = 10,
+  unit = "days",
+}: GetDesksServiceParams): Promise<DeskDto[]> {
+  const desksResult: DeskDto[] = [];
+
+  if (Array.isArray(deskIds) && deskIds.length > 0) {
+    for (const deskId of deskIds) {
+      let stats = await getStats(zoneId, deskId, count, unit)[0];
+      let latestDataPoint = await getLatestDataPoint(
+        zoneId,
+        deskId,
+        count,
+        unit,
+      )[0];
+      let latestActiveDataPoint = await getLatestActiveDataPoint(
+        zoneId,
+        deskId,
+        count,
+        unit,
+      )[0];
+      desksResult.push(
+        mapToDeskDto(stats, latestDataPoint, latestActiveDataPoint),
+      );
+    }
+    return desksResult;
+  } else {
+    const allDesks = await getDesksFromCsv();
+    const allZoneDesks = allDesks.filter((i) => i.zoneId === zoneId);
+    for (const desk of allZoneDesks) {
+      let stats = await getStats(zoneId, desk.deskId, count, unit)[0];
+      let latestDataPoint = await getLatestDataPoint(
+        zoneId,
+        desk.deskId,
+        count,
+        unit,
+      )[0];
+      let latestActiveDataPoint = await getLatestActiveDataPoint(
+        zoneId,
+        desk.deskId,
+        count,
+        unit,
+      )[0];
+      desksResult.push(
+        mapToDeskDto(stats, latestDataPoint, latestActiveDataPoint),
+      );
+    }
+    return desksResult;
+  }
```

Comment:
This doesn't work. The promises from InfluxService are not resolved before `mapToDeskDto` is called => error because undefined params